lib/ansible/module_utils/basic.py

```diff
@@ -128,18 +128,20 @@
     for attribute in ('available_algorithms', 'algorithms'):
         algorithms = getattr(hashlib, attribute, None)
         if algorithms:
+            # convert algorithms to list instead of immutable tuple so md5 can be removed if not available
+            algorithms = list(algorithms)
             break
     if algorithms is None:
         # python 2.5+
-        algorithms = ('md5', 'sha1', 'sha224', 'sha256', 'sha384', 'sha512')
+        algorithms = ['md5', 'sha1', 'sha224', 'sha256', 'sha384', 'sha512']
     for algorithm in algorithms:
         AVAILABLE_HASH_ALGORITHMS[algorithm] = getattr(hashlib, algorithm)
 
     # we may have been able to import md5 but it could still not be available
     try:
         hashlib.md5()
     except ValueError:
-        algorithms.pop('md5', None)
+        algorithms.remove('md5')
```

Comment:
You've definitely identified a bug here, however, I think the bug is slightly different than what you're solving.

Reading this code, I think that ```algorithms``` is a private variable that's just used to calculate ```AVAILABLE_HASH_ALGORITHMS```.  So it is fine to leave ```algorithms``` as a tuple above.  Instead, this line, 144 should be modifying ```AVAILABLE_HASH_ALGORITHMS```:

```suggestion
        AVAILABLE_HASH_ALGORITHMS.pop('md5', None)
```

It took me a while to see what was intended to happen here but eventually, seeing that the precise invocation of pop() didn't work with either a list or a tuple I figured out that there was something else going on here.  I then saw that in the fallback methods below we were not setting ```algorithms```, only ```AVAILABLE_HASH_ALGORITHMS``` and that lead me to see what the code meant to do.